### PR TITLE
Chore: Account deletion via API

### PR DESCRIPTION
**Pull request by Rory-Powell**
## Description

Updates: 
- Allow tenant deletion api to work when internal api key is supplies

See: https://github.com/Budibase/account-portal/pull/292

Addresses: 
- [BUDI-6794](https://linear.app/budibase/issue/BUDI-6794/account-deletion-via-api)

### CODE REVIEW

File: packages/worker/src/api/controllers/system/tenants.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ export async function destroy(ctx: UserCtx) {
   const user = ctx.user!
   const tenantId = ctx.params.tenantId
 
-  if (tenantId !== user.tenantId) {
+  if (!ctx.internal && tenantId !== user.tenantId) {
```
Comment (adrinr) [inline packages/worker/src/api/controllers/system/tenants.ts:8]: Are we using user tokens for internal calls?

Comment (Rory-Powell) [reply packages/worker/src/api/controllers/system/tenants.ts:8]: we use the `INTERNAL_API_KEY`

**Review by shogunpurple — APPROVED** [review]
LGTM!

File: packages/worker/src/api/controllers/system/tenants.ts (hunk 1 above)
Comment (Rory-Powell) [reply packages/worker/src/api/controllers/system/tenants.ts:8]: In this case we want to only allow the action if the current session matches the users tenant id, or the api key is the internal api (system) api key

**Pull request merged**